gpt_index/indices/query/graph_query_engine.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any, Dict, List, Optional, Tuple
+from gpt_index.data_structs.node_v2 import IndexNode, Node, NodeWithScore
+from gpt_index.indices.postprocessor.node import BaseNodePostprocessor
+from gpt_index.indices.query.base import BaseQueryEngine
+from gpt_index.indices.query.schema import QueryBundle
+from gpt_index.indices.response.response_synthesis import ResponseSynthesizer
+from gpt_index.response.schema import RESPONSE_TYPE
+
+
+class ComposableGraphQueryEngine(BaseQueryEngine):
+    def __init__(
+        self,
+        graph: Any,
```

Comment:
what if the query engine also took in the retriever classes for each index? that way the user can customize which retriever classes they'd want to pass in. reduces reliance on `as_retriever` below

Reply:
Revised to do this.